tests/test_query_parsing.py

```diff
@@ -219,30 +259,29 @@ async def test_approximate_query_as_subquery(self):
     register_inference_services(aidb_engine, data_dir)
 
     config = aidb_engine._config
-
-    # approx aggregation as a subquery
-    query_str1 = '''
-                 SELECT x_min FROM  objects00
-                 WHERE x_min > (SELECT AVG(x_min) FROM objects00 ERROR_TARGET 10% CONFIDENCE 95%)
-                 AND table2.object_id > (SELECT AVG(ob.object_id) FROM objects00 AS ob WHERE frame > 500);
-                 '''
-
-    parsed_query = Query(query_str1, config)
-    with self.assertRaises(Exception):
-      _ = parsed_query.all_queries_in_expressions
-
-    # approx select as a subquery
-    query_str2 = '''
-                 SELECT x_min FROM  objects00
-                 WHERE frame IN (SELECT frame FROM colors02 where color LIKE 'blue'
-                    RECALL_TARGET {RECALL_TARGET}% CONFIDENCE 95%;)
-                 AND table2.object_id > (SELECT AVG(ob.object_id) FROM objects00 AS ob WHERE frame > 500);
-                 '''
-
-    parsed_query = Query(query_str2, config)
+    queries = {
+      # approx aggregation as a subquery
+      "query_str1": 
+          '''
+          SELECT x_min 
+          FROM  objects00
+          WHERE x_min > (SELECT AVG(x_min) FROM objects00 ERROR_TARGET 10% CONFIDENCE 95%)
+          AND table2.object_id > (SELECT AVG(ob.object_id) FROM objects00 AS ob WHERE frame > 500);
+          ''',
+      # approx select as a subquery
+      "query_str2": 
+          '''
+          SELECT x_min 
+          FROM  objects00
+          WHERE frame IN (SELECT frame FROM colors02 where color LIKE 'blue'
+          RECALL_TARGET {RECALL_TARGET}% CONFIDENCE 95%;)
+          AND table2.object_id > (SELECT AVG(ob.object_id) FROM objects00 AS ob WHERE frame > 500);
+          '''
+    }
+    parsed_query = Query(queries["query_str1"]+queries["query_str2"], config)
```

Comment:
why you add these two string together?

Reply:
The intent was to test to merge them into one check as the assert is raised even if one of them does not contain approximate query as a subquery. Please let me know if we need a loop instead.

Reply:
use a loop

Reply:
Done, used a loop.